src/effects/EffectUI.cpp

```diff
@@ -554,12 +554,10 @@ void EffectUIHost::OnApply(wxCommandEvent & evt)
       return;
    }
    
-   // This will take care of calling TransferDataFromWindow() for an effect.
-   if (!mEffectUIHost.GetDefinition().SaveUserPreset(
-      mEffectUIHost.GetCurrentSettingsGroup(), mpAccess->Get()))
-   {
+   if (!TransferDataFromWindow() ||
+       !mEffectUIHost.GetDefinition().SaveUserPreset(
+         mEffectUIHost.GetCurrentSettingsGroup(), mpAccess->Get()))
       return;
```

Comment:
Just a general consideration about these methods: they return void, so if TransferDataFromWindow fails, the error is not propagated to the caller.

Reply:
TransferDataFromWindow may fail because a number was typed that is out of the allowed range, for example.  It may alert the user with an error dialog.